repos/system_upgrade/el7toel8/actors/bindupdate/libraries/updates.py

```diff
@@ -0,0 +1,85 @@
+from leapp.libraries.common import isccfg
+from leapp.libraries.stdlib import run, api
+
+# Callback for walk function
+callbacks = {
+    'dnssec-lookaside': isccfg.ModifyState.callback_comment_out,
+}
+
+
+def paths_from_issues(issues):
+    """Extract paths from list of BindConfigIssuesModel."""
+    paths = []
+    for issue in issues:
+        paths.append(issue.path)
+    return paths
+
+
+def parser_file(parser, path):
+    for cfg in parser.FILES_TO_CHECK:
+        if cfg.path == path:
+            return cfg
+    return None
+
+
+def make_backup(path, backup_suffix='.leapp'):
+    """Make backup of a file before modification."""
+    backup_path = path + backup_suffix
+    run(['cp', '--preserve=all', path, backup_path])
```

Comment:
This can potentially fail and raise CommandError. Please wrap this in try/except

Reply:
If the backup fails, some error has to be raised. If even making backup copy fails, I do not think it should continue updating anything. How should be failed actor marked? Is there correct error to be raised? I expect actor-level try/except should be used, is it?

Reply:
@pemensik  actor should never crash in the hard way. in the worst valid case it should raise `StopActorExecutionError` exception with proper error msg and possible details explanations (that exception has additional special handler on the framework lvl)

Reply:
Sorry for late response, missed the ping (twice!). How about the usual approach we take, by catching CallesProcessError and raising StopActorExecutionError (if you believe that there is no sense in proceeding)?

```
from leapp.exceptions import StopActorExecutionError
from leapp.libraries.stdlib import api, CalledProcessError, run

try:
    run(['cp', '--preserve=all', path, backup_path])
except CalledProcessError as exc:
    raise StopActorExecutionError('Could not create a backup copy', 'details': 'An exception during backup raised {}'.format(str(exc)))
```